plugin/src/main/scala/eu/unicredit/swagger/FileManagement.scala

```diff
@@ -18,9 +18,8 @@ import java.io.File
 
 object FolderCreator {
 
-  def genPackage(base: String, packageName: String): File = {
-    val packagePath = base + File.separator + packageName.replace(".", File.separator)
-    val packageDir = new File(packagePath)
+  def genPackage(base: File, packageName: String): File = {
+    val packageDir = new File(base, packageName.replace(".", File.separator))
     packageDir.mkdirs()
```

Comment:
Since you are refactoring a bit I would suggest some more :-)
avoid here to mkDir and instead use http://www.scala-sbt.org/0.13/api/#sbt.IO$ IO to proper write files.